tutorials/dataframe/df103_NanoAODHiggsAnalysis.py

```diff
@@ -0,0 +1,574 @@
+
+    df_h_sig_4mu = df_sig_4mu_reco\
+        .Define("weight", "{}".format(weight_sig_4mu))\
+        .Histo1D(("h_sig_4mu", "", nbins, 70, 180), "H_mass", "weight")
+
+    df_bkg_4mu_reco = reco_higgs_to_4mu(df_bkg_4mu)
+
+    df_h_bkg_4mu = df_bkg_4mu_reco\
+        .Define("weight", "{}".format(weight_bkg_4mu))\
+        .Histo1D(("h_bkg_4mu", "", nbins, 70, 180), "H_mass", "weight")
+
+    df_data_4mu_reco = reco_higgs_to_4mu(df_data_doublemu)
+
+    df_h_data_4mu = df_data_4mu_reco\
+        .Define("weight", "1.0")\
+        .Histo1D(("h_data_4mu", "", nbins, 70, 180), "H_mass", "weight")
+
+    # Reconstruct Higgs to 4 electrons
+    df_sig_4el_reco = reco_higgs_to_4el(df_sig_4l)
+
+    df_h_sig_4el = df_sig_4el_reco\
+        .Define("weight", "{}".format(weight_sig_4el))\
+        .Histo1D(("h_sig_4el", "", nbins, 70, 180), "H_mass", "weight")
+
+    df_bkg_4el_reco = reco_higgs_to_4el(df_bkg_4el)
+
+    df_h_bkg_4el = df_bkg_4el_reco\
+        .Define("weight", "{}".format(weight_bkg_4el))\
+        .Histo1D(("h_bkg_4el", "", nbins, 70, 180), "H_mass", "weight")
+
+    df_data_4el_reco = reco_higgs_to_4el(df_data_doubleel)
+
+    df_h_data_4el = df_data_4el_reco\
+        .Define("weight", "1.0")\
+        .Histo1D(("h_data_4el", "", nbins, 70, 180), "H_mass", "weight")
+
+    # Reconstruct Higgs to 2 electrons and 2 muons
+    df_sig_2el2mu_reco = reco_higgs_to_2el2mu(df_sig_4l)
+
+    df_h_sig_2el2mu = df_sig_2el2mu_reco\
+        .Define("weight", "{}".format(weight_sig_2el2mu))\
+        .Histo1D(("h_sig_2el2mu", "", nbins, 70, 180), "H_mass", "weight")
+
+    df_bkg_2el2mu_reco = reco_higgs_to_2el2mu(df_bkg_2el2mu)
+
+    df_h_bkg_2el2mu = df_bkg_2el2mu_reco\
+        .Define("weight", "{}".format(weight_bkg_2el2mu))\
+        .Histo1D(("h_bkg_2el2mu", "", nbins, 70, 180), "H_mass", "weight")
+
+    df_data_2el2mu_reco = reco_higgs_to_2el2mu(df_data_doublemu)
+
+    df_h_data_2el2mu = df_data_2el2mu_reco\
+        .Define("weight", "1.0")\
+        .Histo1D(
+            ("h_data_2el2mu_doublemu", "", nbins, 70, 180),
+            "H_mass", "weight"
+        )
+
+    # Trigger event loops and retrieve histograms
+    signal_4mu = df_h_sig_4mu.GetValue()
+    background_4mu = df_h_bkg_4mu.GetValue()
+    data_4mu = df_h_data_4mu.GetValue()
+
+    signal_4el = df_h_sig_4el.GetValue()
+    background_4el = df_h_bkg_4el.GetValue()
+    data_4el = df_h_data_4el.GetValue()
+
+    signal_2el2mu = df_h_sig_2el2mu.GetValue()
+    background_2el2mu = df_h_bkg_2el2mu.GetValue()
+    data_2el2mu = df_h_data_2el2mu.GetValue()
+
+    # Make plots
+    plot(
+        signal_4mu, background_4mu, data_4mu,
+        "m_{4#mu} (GeV)", "higgs_4mu.pdf"
+    )
+    plot(
+        signal_4el, background_4el, data_4el,
+        "m_{4e} (GeV)", "higgs_4el.pdf"
+    )
+    plot(
+        signal_2el2mu, background_2el2mu, data_2el2mu,
+        "m_{2e2#mu} (GeV)", "higgs_2el2mu.pdf"
+    )
+
+    # Final datasets
```

Comment:
I would not say `Final datasets` but `Final plot` or `Combined plot`.

Reply:
Ok!